Bottom: b17b01e240e19a71bb68f68b9a34d20804ac3414
Top:    82d75360516ef40b2cf28c1fbe7eba234c3fa091
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-09 17:30:22 +0000

Add HID_DRAW* parameter to (HID_DRAW *)->{flush,finish}_debug_draw_gc() function


---

diff --git a/src/autoroute.c b/src/autoroute.c
index 9ca83f4..4c21f68 100644
--- a/src/autoroute.c
+++ b/src/autoroute.c
@@ -3740,7 +3740,7 @@ TracePath (routedata_t * rd, routebox_t * path, const routebox_t * target,
 
 #ifdef ROUTE_DEBUG
   if (ddraw != NULL)
-    gui->flush_debug_draw ();
+    gui->flush_debug_draw (ddraw);
 #endif
 }
 
@@ -5408,7 +5408,7 @@ donerouting:
   if (ddraw != NULL)
     {
       hid_draw_destroy_gc (ar_gc);
-      gui->finish_debug_draw ();
+      gui->finish_debug_draw (ddraw);
     }
 #endif
 
diff --git a/src/hid.h b/src/hid.h
index 5ecf529..580e47e 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -472,13 +472,13 @@ typedef enum
      * May be implemented as a NOOP if the GUI has chosen to send the
      * debug drawing directly to the screen.
      */
-    void (*flush_debug_draw)   (void);
+    void (*flush_debug_draw)   (HID_DRAW *hid_draw);
 
     /* When finished, the user must inform the GUI to clean up resources
      *
      * Any remaining rendering will be flushed to the screen.
      */
-    void (*finish_debug_draw)  (void);
+    void (*finish_debug_draw)  (HID_DRAW *hid_draw);
 
     /* Notification to the GUI around saving the PCB file.
      *
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index f4b8d2d..f82c498 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -425,12 +425,12 @@ nogui_request_debug_draw (void)
 }
 
 static void
-nogui_flush_debug_draw (void)
+nogui_flush_debug_draw (HID_DRAW *hid_draw)
 {
 }
 
 static void
-nogui_finish_debug_draw (void)
+nogui_finish_debug_draw (HID_DRAW *hid_draw)
 {
 }
 
diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 2a2a280..6bbd1b0 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -1314,16 +1314,16 @@ ghid_request_debug_draw (void)
 }
 
 void
-ghid_flush_debug_draw (void)
+ghid_flush_debug_draw (HID_DRAW *hid_draw)
 {
   ghid_screen_update ();
   gdk_flush ();
 }
 
 void
-ghid_finish_debug_draw (void)
+ghid_finish_debug_draw (HID_DRAW *hid_draw)
 {
-  ghid_flush_debug_draw ();
+  ghid_flush_debug_draw (gc);
   /* No special tear down requirements
    */
 }
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 279ccc3..d4b0598 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -3023,7 +3023,7 @@ ghid_request_debug_draw (void)
 }
 
 void
-ghid_flush_debug_draw (void)
+ghid_flush_debug_draw (HID_DRAW *hid_draw)
 {
   render_priv *priv = gport->render_priv;
   GtkWidget *widget = gport->drawing_area;
@@ -3038,7 +3038,7 @@ ghid_flush_debug_draw (void)
 }
 
 void
-ghid_finish_debug_draw (void)
+ghid_finish_debug_draw (HID_DRAW *hid_draw)
 {
   render_priv *priv = gport->render_priv;
 
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index f7597f5..a8c6c36 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -508,8 +508,8 @@ gboolean ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * ev);
 GdkPixmap *ghid_render_pixmap (int cx, int cy, double zoom,
                                int width, int height, int depth);
 HID_DRAW *ghid_request_debug_draw (void);
-void ghid_flush_debug_draw (void);
-void ghid_finish_debug_draw (void);
+void ghid_flush_debug_draw (HID_DRAW *hid_draw);
+void ghid_finish_debug_draw (HID_DRAW *hid_draw);
 bool ghid_event_to_pcb_coords (int event_x, int event_y, Coord *pcb_x, Coord *pcb_y);
 bool ghid_pcb_to_event_coords (Coord pcb_x, Coord pcb_y, int *event_x, int *event_y);
 void ghid_port_rotate (void *ball, float *quarternion, gpointer userdata);
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 83fbd55..4ed4237 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -4038,7 +4038,7 @@ lesstif_request_debug_draw (void)
 }
 
 static void
-lesstif_flush_debug_draw (void)
+lesstif_flush_debug_draw (HID_DRAW *hid_draw)
 {
   /* Copy the backing pixmap to the display and redraw any attached objects */
   XSetFunction (display, my_gc, GXcopy);
@@ -4054,9 +4054,9 @@ lesstif_flush_debug_draw (void)
 }
 
 static void
-lesstif_finish_debug_draw (void)
+lesstif_finish_debug_draw (HID_DRAW *hid_draw)
 {
-  lesstif_flush_debug_draw ();
+  lesstif_flush_debug_draw (hid_draw);
   /* No special tear down requirements
    */
 }
